app/model/block.py

```diff
@@ -10,11 +10,73 @@ def __init__(self):
         self.container = None
         self.questionnaire = None
         self.validation = None
-        self.questionniare = None
+        self.questionnaire = None
         self.templatable_properties = []
         self.display = Display()
 
     def add_section(self, section):
         if section not in self.sections:
             self.sections.append(section)
             section.container = self
+
+    def to_json(self):
+        json_dict = {
+            "id": self.id,
+            "title": self.title,
+            "sections": [],
+            "validation": [],
+            "display": {}
+        }
+
+        for section in self.sections:
+            json_dict['sections'].append(section.to_json())
+
+        if self.validation is not None:
+            for validation in self.validation:
+                json_dict['validation'].append(validation.to_json())
+
+        if self.display is not None:
+            json_dict['display'] = self.display.to_json()
+
+        return json_dict
+
+    def __eq__(self, other):
```

Comment:
If you stick with the __eq__ approach also check __hash__ e.g. https://docs.python.org/3/reference/datamodel.html#object.__hash__ and I believe (but could be wrong that __ne__ by default calls __eq__ and inverts the result (i.e. no need to explicitly add it). Also consider better unit tests to cover the operators.